pkgs/servers/typedb/default.nix

```
@@ -0,0 +1,99 @@
+  };
+  linuxSrcUrl = fetchurl {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+    sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
+  };
+  windowsSrc = fetchzip {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-windows-2.1.1.zip";
+    sha256 = "0vd66gfshkg697z07nhy957mwqzlli4r4pmn67hx58n9mkg024kq";
+  };
+  macSrc = fetchzip {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-mac-2.1.1.zip";
+    sha256 = "16hlfy6kh2rnvcralz206q13mghb0rv8wazpg6q3h324p5rdys54";
+  };
+  srcFolder = if stdenv.hostPlatform.isWindows then windowsSrc
+              else if stdenv.isDarwin          then macSrc
+                                               else linuxSrcUrl ;
+  javaPatch = ''
+        20c20
+        < JAVA_BIN=java
+        ---
+        > JAVA_BIN=${openjdk}/bin/java
+        '';
+
+in
+stdenv.mkDerivation rec {
+  pname = "typedb";
+  version = typedbVersion;
+
+  src = srcFolder;
+    
+  phases = [ "installPhase" "installCheckPhase" ];
+
+  buildDepends = [ openjdk ];
+
+  installPhase = ''
+    ls
+    tar -xf typedb-all-linux-2.1.1.tar.gz
+    echo "here"
+    ls -lah 
+    echo "--"
+    #patch before install
+    echo "${javaPatch}" > typedb_java.patch
+    patch ./${typedbDir}/typedb typedb_java.patch
+
+    mkdir $out
+    cp -r ./${typedbDir} $out
+    # add a wrapper script to $out that will move typedb to $typedb
+    # this is necessary because typedb needs a writable environment
+    echo "
+      # on the first start copy everything to typedbHome
+      if [ ! -f ${typedbHome}/typedb ]; then 
+        mkdir -p ${typedbHome}; 
+        cp -r $out/${typedbDir}/* ${typedbHome}; 
+    
+        # correct permissions so that typedb and the user can write there 
+        chmod -R u+rw ${typedbHome}
+        chmod u+x ${typedbHome}/typedb
+      fi; 
+      ${typedbHome}/typedb \$@; 
+    " > $out/typedb
+
+    chmod +x $out/typedb
+  '';
+
+  doInstallCheck = true;
+  installCheckPhase= ''
+      $out/typedb --help > /dev/null
+      res=`$out/typedb client`
+      echo "--"
+      echo $res
+      '';
+  doCheck = true;
+
+  meta = with lib; {
+    description = "TypeDB is a distributed knowledge graph: a logical database to organise large and complex networks of data as one body of knowledge.";
+    longDescription = ''
+        TypeDB is a distributed knowledge graph: a logical database to organise large and complex networks of data as one body of knowledge. 
+        TypeDB provides the knowledge engineering tools for developers to easily leverage the power of Knowledge Representation and Automated 
+        Reasoning when building complex systems. Ultimately, TypeDB serves as the knowledge-base foundation for intelligent systems.
+    '';
+    homepage = "https://www.grakn.ai/";
+    license = licenses.gpl3Plus;
+    platforms = platforms.all;
```

```suggestion
    platforms = lib.attrNames typedbDirs;
```

Or perhaps it's nicer to have a single attrset of systems containing in each attribute an attrset with the dir and the src. That way you don't duplicate the list of systems and you don't have to choose arbitrarily here.